etl/steps/data/garden/rff/2025-05-19/world_carbon_pricing.py

```diff
@@ -0,0 +1,299 @@
+"""Create a garden dataset from the meadow dataset."""
+
+from typing import Dict, List
+
+from owid.catalog import Table, Variable
+from owid.datautils import dataframes, io
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# It may happen that the data for the most recent year is incomplete.
+# If so, define the following to be last year fully informed.
+# LAST_INFORMED_YEAR = 2021
+LAST_INFORMED_YEAR = None
+
+# Labels for the variables showing whether any sector is covered by an ETS or a carbon tax at the national or only
+# sub-national level.
+LABEL_ETS_NOT_COVERED = "No ETS"
+LABEL_ETS_COVERED = "Has an ETS"
+LABEL_ETS_COVERED_ONLY_SUBNATIONAL = "Has an ETS only at a sub-national level"
+LABEL_TAX_NOT_COVERED = "No carbon tax"
+LABEL_TAX_COVERED = "Has a carbon tax"
+LABEL_TAX_COVERED_ONLY_SUBNATIONAL = "Has a carbon tax only at a sub-national level"
+# If a country-years has both national and sub-national coverage, mention only the national and ignore subnational.
+LABEL_ETS_COVERED_NATIONAL_AND_SUBNATIONAL = "Has an ETS"
+LABEL_TAX_COVERED_NATIONAL_AND_SUBNATIONAL = "Has a carbon tax"
+
+# Columns to keep from raw dataset and how to rename them.
+COLUMNS = {
+    "jurisdiction": "country",
+    "year": "year",
+    "ipcc_code": "ipcc_code",
+    "product": "product",
+    "sector_name": "sector_name",
+    "tax": "tax",
+    "ets": "ets",
+    "tax_rate_excl_ex_clcu": "tax_rate_gross",
+    "tax_rate_incl_ex_clcu": "tax_rate_net",
+    "ets_price": "ets_price",
+}
+
+# Mapping of countries and the regions of the country included in the sub-national dataset.
+# In the future, it would be good to load this mapping as additional data (however, the mapping is hardcoded in the
+# original repository, so it's not trivial to get this mapping automatically).
+COUNTRY_MEMBERS_FILE = paths.directory / "world_carbon_pricing.country_members.json"
```

Comment:
Is it worth asserting each country has the expected number of members?

Reply:
The code raises a warning if there are missing or unused mappings.